packages/truffle-core/lib/console.js

```diff
@@ -1,248 +1,225 @@
+      return this.command.run(cmd.trim(), this.options, error => {
+        if (error) {
+          // Perform error handling ourselves.
+          if (error instanceof TruffleError) {
+            console.log(error.message);
+          } else {
+            // Bubble up all other unexpected errors.
+            console.log(error.stack || error.toString());
+          }
+          return callback();
         }
-        return callback();
-      }
-
-      // Reprovision after each command as it may change contracts.
-      self.provision(function(err) {
-        // Don't pass abstractions to the callback if they're there or else
-        // they'll get printed in the repl.
-        callback(err);
+
+        // Reprovision after each command as it may change contracts.
+        this.provision()
+          .then(() => callback())
+          .catch(error => {
+            // Don't pass abstractions to the callback if they're there or else
+            // they'll get printed in the repl.
+            callback(error);
+          });
       });
-    });
-  }
+    }
 
-  // Much of the following code is from here, though spruced up:
-  // https://github.com/nfcampos/await-outside
-
-  /*
-  - allow whitespace before everything else
-  - optionally capture `var|let|const <varname> = `
-    - varname only matches if it starts with a-Z or _ or $
-      and if contains only those chars or numbers
-    - this is overly restrictive but is easier to maintain
-  - capture `await <anything that follows it>`
-  */
-  let includesAwait = /^\s*((?:(?:var|const|let)\s+)?[a-zA-Z_$][0-9a-zA-Z_$]*\s*=\s*)?(\(?\s*await[\s\S]*)/;
-
-  var match = cmd.match(includesAwait);
-  var source = cmd;
-  var assignment = null;
-
-  // If our code includes an await, add special processing to ensure it's evaluated properly.
-  if (match) {
-    var assign = match[1];
-    var expression = match[2];
-
-    var RESULT = "__await_outside_result";
-
-    // Wrap the await inside an async function.
-    // Strange indentation keeps column offset correct in stack traces
-    source = `(async function() { try { ${
-      assign ? `global.${RESULT} =` : "return"
-    } (
-${expression.trim()}
-); } catch(e) { global.ERROR = e; throw e; } }())`;
-
-    assignment = assign
-      ? `${assign.trim()} global.${RESULT}; void delete global.${RESULT};`
-      : null;
-  }
+    // Much of the following code is from here, though spruced up:
+    // https://github.com/nfcampos/await-outside
+
+    /*
+    - allow whitespace before everything else
+    - optionally capture `var|let|const <varname> = `
+      - varname only matches if it starts with a-Z or _ or $
+        and if contains only those chars or numbers
+      - this is overly restrictive but is easier to maintain
+    - capture `await <anything that follows it>`
+    */
+    let includesAwait = /^\s*((?:(?:var|const|let)\s+)?[a-zA-Z_$][0-9a-zA-Z_$]*\s*=\s*)?(\(?\s*await[\s\S]*)/;
+
+    var match = cmd.match(includesAwait);
```

Comment:
Can these be `let` or am I missing something re: the scope?

Reply:
Yeah I don't see any reason these can't be updated!